webpack/src/Components/Environments/Welcome.js

```diff
@@ -10,16 +11,34 @@ export const WelcomeEnv = ({ canCreate }) => {
     url: foremanUrl('/foreman_puppet/environments/new'),
   };
 
-  const content = __(`If you are planning to use Foreman as an external node classifier you should provide information about one or more environments.<br/>
-  This information is commonly imported from a pre-existing Puppet configuration by the use of the <a target="_blank" href=${getManualURL(
-    '4.2.2Classes'
-  )}>Puppet classes and environment importer.</a>`);
+  const description = (
+    <>
+      <FormattedMessage
+        id="puppetenv-welcome"
+        defaultMessage={__(
+          'If you are planning to use Foreman as an external node classifier you should provide information about one or more environments.{newLine}This information is commonly imported from a pre-existing Puppet configuration by the use of the {puppetClasses}.'
```

Comment:
Late to the party, so sorry about that.

You have improved it, but you still have made something that's very hard to translate because you string parts of a sentence together. In a language where the word order is not the same it may be incomprehensible for the reader.

What I think you should probably do is have a `{link-start}` and `{link-end}` placeholder so translators have the freedom to move it around.

Reply:
Oh, and sadly we can't extract multi-line strings now so this string doesn't show up in the .pot file.